Lib/concurrent/futures/thread.py

```diff
@@ -80,7 +80,14 @@ def _worker(executor_reference, work_queue, initializer, initargs):
                 work_item.run()
                 # Delete references to object. See issue16284
                 del work_item
+
+                # attempt to increment idle count
+                executor = executor_reference()
+                if executor is not None:
+                    executor._increase_idle_count()
```

Comment:
If `_idle_count` is a semaphore, you just need to increase the number with `release`:
```
executor._idle_count.release()
```